syntax = "proto3";
package protos.playbooks;

import "google/protobuf/wrappers.proto";

message Sentry {
  message FetchIssueInfoById {
    google.protobuf.StringValue issue_id = 1;
  }

  message FetchProjectEvents {
    google.protobuf.StringValue project_slug = 1;
  }

  message FetchRecentErrors {
    google.protobuf.StringValue project_slug = 1;
  }

  message FetchEventInfoById {
    google.protobuf.StringValue event_id = 1;
    google.protobuf.StringValue project_slug = 2;
  }

  message FetchListOfRecentEventsWithSearchQuery{
    google.protobuf.StringValue project_slug = 1;
    google.protobuf.StringValue query = 2;
    google.protobuf.Int64Value max_events_to_analyse = 3;
  }

  enum TaskType {
    UNKNOWN = 0;
    FETCH_ISSUE_INFO_BY_ID = 1;
    FETCH_PROJECT_EVENTS = 2;
    FETCH_RECENT_ERRORS = 3;
    FETCH_EVENT_INFO_BY_ID = 4;
    FETCH_LIST_OF_RECENT_EVENTS_WITH_SEARCH_QUERY = 5;
  }

  TaskType type = 1;
  oneof task {
    FetchIssueInfoById fetch_issue_info_by_id = 2;
    FetchProjectEvents fetch_project_events = 3;
    FetchRecentErrors fetch_recent_errors = 4;
    FetchEventInfoById fetch_event_info_by_id = 5;
    FetchListOfRecentEventsWithSearchQuery fetch_list_of_recent_events_with_search_query = 6;
  }
}